core/src/main/java/org/apache/iceberg/SchemaParser.java

```diff
@@ -144,7 +155,7 @@ static void toJson(Type type, JsonGenerator generator) throws IOException {
   }
 
   public static void toJson(Schema schema, JsonGenerator generator) throws IOException {
-    toJson(schema.asStruct(), schema.schemaId(), generator);
+    toJson(schema.asStruct(), schema.schemaId(), schema.rowIdentifiers(), generator);
```

Comment:
not sure why this method was not updated in the past, is there any risk to use the v2 json writer?